pkg/kn/commands/logger/logger.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+type Config struct {
+	QuietMode bool
+}
+
+func NewLogger(config Config) io.Writer {
```

Comment:
Instead of returning a writer, I probably would initialize a global variable here, that stores the write, so that subsequent calls to `Log` would need to pass back in the Writer again. E.g rename this just to `func InitLogger(conf Config) { ...` (without a return value) then store a static variable `outWriter` and use that `outWriter` in the `Log()` method directly.